components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java

```diff
@@ -3175,10 +3175,13 @@ public Response createNewAPIVersion(String newVersion, String apiId, Boolean def
                 throw new APIMgtResourceNotFoundException("API not found for id " + apiId,
                         ExceptionCodes.from(ExceptionCodes.API_NOT_FOUND, apiId));
             }
-            if (newVersion.equals(existingAPI.getId().getVersion())) {
-                throw new APIMgtResourceAlreadyExistsException("Version " + newVersion + " exists for api "
-                        + existingAPI.getId().getApiName(), ExceptionCodes.from(API_VERSION_ALREADY_EXISTS, newVersion,
-                            existingAPI.getId().getApiName()));
+            //Get all existing versions of API
+            List<String> apiVersions = apiProvider.getApiVersionsMatchingApiNameAndOrganization(
+                    apiIdentifierFromTable.getApiName(), apiIdentifierFromTable.getProviderName(), organization);
+            if (apiVersions.contains(newVersion)) {
+                throw new APIMgtResourceAlreadyExistsException(
+                        "Version " + newVersion + " exists for api " + existingAPI.getId().getApiName(),
+                        ExceptionCodes.from(API_VERSION_ALREADY_EXISTS, newVersion, existingAPI.getId().getApiName()));
```

Comment:
Have we added the relevant changes in integration test to support this, please attach the above PR as well.

Reply:
Done